### PR TITLE
fix(gauge): value arc angle computation

### PR DESCRIPTION
**Pull request by skjindal93**
Fixes https://github.com/hypertrace/hypertrace-ui/pull/486#discussion_r554518546

### CODE REVIEW

File: projects/observability/src/shared/components/gauge/gauge.component.ts [hunk 1]
```diff
@@ -104,11 +104,13 @@ export class GaugeComponent implements OnChanges {
...
 
   private buildValueArc(radius: number, inputData: GaugeInputData): string {
+    const completeAngle = Math.PI / 2 + GaugeComponent.EXTRA_ARC_ANGLE;
+
     return this.buildArcGenerator()({
       innerRadius: radius - GaugeComponent.GAUGE_RING_WIDTH,
       outerRadius: radius,
-      startAngle: -Math.PI / 2 - GaugeComponent.EXTRA_ARC_ANGLE,
-      endAngle: -Math.PI / 2 - GaugeComponent.EXTRA_ARC_ANGLE + (inputData.value / inputData.maxValue) * Math.PI
+      startAngle: -completeAngle,
```
Comment (aaron-steinfeld) [inline projects/observability/src/shared/components/gauge/gauge.component.ts:112]: Does this handle the first case described in the comment? If inputData.value is 0, we're doing `[-completeAngle, -completeAngle]` so there'd still be no arc

Comment (skjindal93) [reply projects/observability/src/shared/components/gauge/gauge.component.ts:112]: Yes. We want no value arc, when the inputData.value is 0. In fact @anandtiwary mentioned that's the expected outcome as well

>case 1: inputData.value = 0, the end angle would be -Math.PI / 2 - GaugeComponent.EXTRA_ARC_ANGLE which is same as start angle. So the arc will be completely empty. Expected outcome.

Comment (aaron-steinfeld) [reply projects/observability/src/shared/components/gauge/gauge.component.ts:112]: Ah I misread his comment - I thought our ux generally has a little arc for a 0-value gauge to show it's "emptyiness"? I'm pretty sure that's what we do in the linear case.

Comment (skjindal93) [reply projects/observability/src/shared/components/gauge/gauge.component.ts:112]: hmm. fair point. I can use a value of 0.5, if the value is 0 to display the 0-value arc

Comment (aaron-steinfeld) [reply projects/observability/src/shared/components/gauge/gauge.component.ts:112]: I think that makes sense - something like `percentFill = Math.min(0.01, inputData.value / inputData.maxValue)`

File: projects/observability/src/shared/components/gauge/gauge.component.ts [hunk 2]
```diff
@@ -110,7 +110,7 @@ export class GaugeComponent implements OnChanges {
       innerRadius: radius - GaugeComponent.GAUGE_RING_WIDTH,
       outerRadius: radius,
       startAngle: -completeAngle,
-      endAngle: -completeAngle + 2 * (inputData.value / inputData.maxValue) * completeAngle
+      endAngle: -completeAngle + 2 * Math.max(0.05, inputData.value / inputData.maxValue) * completeAngle
```
Comment (aaron-steinfeld) [inline projects/observability/src/shared/components/gauge/gauge.component.ts:113]: `.05` feels high here, how does it look? It's showing a 5% fill in the empty case - I was thinking a 1% fill would be sufficient.

Comment (skjindal93) [reply projects/observability/src/shared/components/gauge/gauge.component.ts:113]: 0.01
Screen Shot 2021-01-11 at 9 56 20 PM

0.02
Screen Shot 2021-01-11 at 9 58 24 PM

0.05
Screen Shot 2021-01-11 at 9 56 42 PM

Comment (skjindal93) [reply projects/observability/src/shared/components/gauge/gauge.component.ts:113]: 0.05 looks high, as you said. Looks like a real value. 

0.02 seems to look good as compared to 0.01

File: projects/observability/src/shared/components/gauge/gauge.component.ts (hunk 1 above)
Comment (anandtiwary) [reply projects/observability/src/shared/components/gauge/gauge.component.ts:112]: I am not sure it looks any better to show a little arc for 0. 

Also, could we rename completeAngle to just startAngle?

Comment (skjindal93) [reply projects/observability/src/shared/components/gauge/gauge.component.ts:112]: > Also, could we rename `completeAngle` to just startAngle and assign it `-completeAngle`

Didn't catch the comment

Comment (anandtiwary) [reply projects/observability/src/shared/components/gauge/gauge.component.ts:112]: `CompleteAngle` is confusing to me. Can we do `const startAngle = - (Math.PI / 2 + GaugeComponent.EXTRA_ARC_ANGLE);` ?

Comment (anandtiwary) [reply projects/observability/src/shared/components/gauge/gauge.component.ts:112]: So, `endAngle =   startAngle + 2 * (inputData.value / inputData.maxValue) * (- startAngle)

**Comment by anandtiwary** [comment]
Anything more needed on this PR? For max risk score, the gauge would not appear full. This seems like an important bug

**Comment by aaron-steinfeld** [comment]
> Anything more needed on this PR? For max risk score, the gauge would not appear full. This seems like an important bug

yes we should fix. I think the consensus was no nub, right @jake-bassett ?

**Comment by skjindal93** [comment]
> > Anything more needed on this PR? For max risk score, the gauge would not appear full. This seems like an important bug
> 
> yes we should fix. I think the consensus was no nub, right @jake-bassett ?

got rid of the nib computation

**Review by aaron-steinfeld — APPROVED** [review]

**Review by anandtiwary — APPROVED** [review]

**Pull request merged**